src/textual/geometry.py

```diff
@@ -129,7 +130,7 @@ def get_distance_to(self, other: Offset) -> float:
         """
         x1, y1 = self
         x2, y2 = other
-        distance = ((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1)) ** 0.5
+        distance = math.sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1))
```

Comment:
mypy complains here because `**` does not necessarily return a float. I changed to use `sqrt`, but can also just `# type: ignore` I suppose.

Reply:
I'd prefer the ignore here. The function call is slower than the operator.

Reply:
Let me begin by saying: I'll (obviously) do what you ultimately decide.

But allow me please to try and make some points:

1. A simple test I ran shows they are equivalent, and if anything then the function call is even faster:

```
~ python3
Python 3.10.8 (main, Oct 13 2022, 09:48:40) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import timeit
>>> x1 = 100
>>> x2 = 90
>>> y1 = 30
>>> y2 = 180
>>> def f1():
...     return ((x2-x1)*(x2-x1) + (y2-y1)*(y2-y1)) ** 0.5
...
>>> def f2():
...     return math.sqrt((x2-x1)*(x2-x1) + (y2-y1)*(y2-y1))
...
>>> import math
>>> timeit.timeit('f1()',globals=globals())
0.19882604101439938
>>> timeit.timeit('f1()',globals=globals())
0.19873045897111297
>>> timeit.timeit('f1()',globals=globals())
0.19961341697489843
>>> timeit.timeit('f1()',globals=globals())
0.19851712498348206
>>> timeit.timeit('f2()',globals=globals())
0.1953267920180224
>>> timeit.timeit('f2()',globals=globals())
0.19321687502088025
>>> timeit.timeit('f2()',globals=globals())
0.19635591603582725
```

2. I may find myself arguing this point later, but generally my humble opinion is that in places that are not insight "tight inner loops", and unless extremely slower, we shouldn't try to optimize such things. I suspect I'll bring this up again in isinstance checks below :)

But as I mentioned -- your call obviously.

(I will say that I prefer `cast(float, ...)` to `# type: ignore`, just because it hides fewer possible issues.

Reply:
You are timing global variables there, which is dwarfing the square root. If you time just the square root, you will find the operator is faster because it doesn't require the additional global and attribute reference.

The operator is a zero cost optimization. If you want to satisfy type checking, try changing the return type to `float | int`.

Reply:
Okay, reverted.